root/static/scripts/edit/externalLinks.js

```diff
@@ -196,6 +196,12 @@ class ExternalLinksEditor extends React.Component<LinksEditorProps, LinksEditorS
               error = l('This relationship type is deprecated and should not be used.');
             } else if ((!isPositiveInteger(link.relationship) || (oldLink && link.url !== oldLink.url)) && checker && !checker(link.url)) {
               error = l('This URL is not allowed for the selected link type, or is incorrectly formatted.');
+            } else if ((!isPositiveInteger(link.relationship) || (oldLink && link.url !== oldLink.url)) && /^(https?:\/\/)?([^.\/]+\.)?wikipedia\.org\/.*#/.test(link.url)) {
+              // Kludge for MBS-9515 to be replaced with the more general MBS-9516
+              error = l('Anchors Wikipedia links are not allowed, please remove {fragment} if still appropriate, see the {url|guidelines}.', {
+                fragment: '<i>' + _.escape(link.url.replace(/^(?:https?:\/\/)?(?:[^.\/]+\.)?wikipedia\.org\/[^#]*#(.*)$/, '#$1')) + '</i>',
```

Comment:
It's much safer to pass `__react: true` here and pass in a React element directly, like `fragment: <i>{link.url.replace(...)}</li>`. Then you can revert the `dangerouslySetInnerHTML` below.

Reply:
Note that for the URL syntax you don't need to pass in an `<a>`, the i18n module will do that for you with `__react: true`. So the `url: {...}` part can stay as it is.

Reply:
Applied in https://github.com/metabrainz/musicbrainz-server/pull/562/commits/46feadc2646625daf81abd03ce12adc529f33f23, with extra style in https://github.com/metabrainz/musicbrainz-server/pull/562/commits/4007afdc64f65f392f1a769eaca344d18773fe37 that wraps the quoted URL fragment in order to break it if longer than line.